src/community/params-extractor/src/main/java/org/geoserver/params/extractor/RequestWrapper.java

```diff
@@ -30,8 +29,11 @@ public RequestWrapper(UrlTransform urlTransform, HttpServletRequest request) {
         super(request);
         this.urlTransform = urlTransform;
         originalParameters = request.getParameterMap();
-        pathInfoPattern = Pattern.compile("^" + request.getContextPath() + "([^/]+?).*$");
-        servletPathPattern = Pattern.compile("^" + request.getContextPath() + "[^/]+?/([^/]+?).*$");
+        pathsPattern =
+                Pattern.compile(
+                        "^"
+                                + request.getContextPath()
+                                + "(/[^\\?^/]+)(/[^\\?]*[^/^\\?])?/?(\\??.*)?$");
```

Comment:
Whoa this regex is evil... I put it into a tool that explains regexes, and still not sure why it's working:

![image](https://user-images.githubusercontent.com/325433/58101461-fa5f8c00-7bdf-11e9-88e1-963c5c393543.png)

The third capturing group seems to be non necessary (nothing using it?), and it would seem like the regex allows for multiple question marks.

Tool here: http://xenon.stanford.edu/~xusch/regexp/

Reply:
Thanks for the awesome tool.  Regular expression improved with non capturing groups and added boorkmark char support (#).